services/spar/test-integration/Test/Spar/Scim/UserSpec.hs

```diff
@@ -622,14 +624,49 @@ testGetUser = do
   storedUser' <- getUser tok (scimUserId storedUser)
   liftIO $ storedUser' `shouldBe` storedUser
 
--- | This is the behavior on develop as well as on the branch where this test was turned on.
--- TODO: find out if this is really what we want, or open an issue and reference the issue
--- here.
-testGetNonScimUser :: TestSpar ()
-testGetNonScimUser = error "finds a non-SCIM-provisioned user and puts it under SCIM management"
+shouldBeManagedBy :: HasCallStack => UserId -> ManagedBy -> TestSpar ()
+shouldBeManagedBy uid flag = do
+  managedBy <- maybe (error "user not found") userManagedBy <$> runSpar (Intra.getBrigUser uid)
+  liftIO $ managedBy `shouldBe` flag
+
+-- | This is (roughly) the behavior on develop as well as on the branch where this test was
+-- turned on.  TODO: find out if this is really what we want, or open an issue and reference
+-- the issue here.
+testGetNonScimSAMLUser :: TestSpar ()
+testGetNonScimSAMLUser = do
+  (_, tid, idp, (_, privcreds)) <- registerTestIdPWithMeta
+  -- NOTE: once SCIM is enabled SSO Auto-provisioning is disabled, so we register the scim token later.
+
+  uidSso <- loginSsoUserFirstTime idp privcreds
+  tok <- registerScimToken tid (Just (idp ^. SAML.idpId))
+
+  shouldBeManagedBy uidSso ManagedByWire
+  _ <- getUser tok uidSso
+  shouldBeManagedBy uidSso ManagedByScim
+
+testGetNonScimInviteUser :: TestSpar ()
+testGetNonScimInviteUser = do
+  brig <- asks (^. teBrig)
+  (tok, (owner, tid, _)) <- registerIdPAndScimToken
+
+  uidNoSso <- userId <$> call (inviteAndRegisterUser brig owner tid)
+  getUser_ (Just tok) uidNoSso brig !!! const 404 === statusCode
 
 testGetUserWithNoHandle :: TestSpar ()
-testGetUserWithNoHandle = error "finds a user that has no handle, and gives it a default handle before responding with it"
+testGetUserWithNoHandle = do
+  (_, tid, idp, (_, privcreds)) <- registerTestIdPWithMeta
+  -- NOTE: once SCIM is enabled SSO Auto-provisioning is disabled, so we register the scim token later.
+  uid <- loginSsoUserFirstTime idp privcreds
```

Comment:
how is this function returning a user without a handle? was that already the behaviour of it?

Reply:
yes